components/datetime/src/format/datetime.rs

```diff
@@ -233,21 +249,25 @@ where
                 field.length,
             )?,
             Year::Cyclic => {
-                w.write_str("(cyclic year: ")?; // TODO(#3761): Add data for cyclic year names
-                format_number(
-                    w,
-                    fixed_decimal_format,
-                    FixedDecimal::from(
-                        datetime
-                            .datetime()
-                            .year()
-                            .ok_or(Error::MissingInputField(Some("year")))?
-                            .cyclic
-                            .ok_or(Error::MissingInputField(Some("cyclic")))?,
-                    ),
-                    field.length,
-                )?;
-                w.write_char(')')?;
+                let datetime = datetime.datetime();
+                let cyclic = datetime
+                    .year()
+                    .ok_or(Error::MissingInputField(Some("year")))?
+                    .cyclic
+                    .ok_or(Error::MissingInputField(Some("cyclic")))?;
+                // TODO(#3761): This is a hack, we should use actual data for cyclic years
+                let cyclics = match datetime.any_calendar_kind() {
+                    Some(AnyCalendarKind::Dangi) => &DANGI_CYCLIC_YEARS,
+                    _ => &CHINESE_CYCLIC_YEARS, // for now assume all other calendars use the stem-branch model
+                };
+                let cyclic_str = usize::try_from(cyclic)
```

Comment:
why is `cyclic` signed? are negative numbers valid? seem like we could avoid the fallibility here

Reply:
no, we can't, because 16-bit platforms exist

`cyclic` probably should not be signed though. we can't change that now

Reply:
We can change that, `u32` has the same size as `i32`, and negative numbers in data (clearly garbage) will just become high numbers.